components/src/jsMain/kotlin/dev/fritz2/components/alert.kt

```diff
@@ -0,0 +1,280 @@
+ * The toast's theme properties are automatically set but can manually be overridden by passing [toastBuild].
+ *
+ * @param styling a lambda expression for declaring the styling of the toast using fritz2's styling DSL
+ * @param baseClass optional CSS class that should be applied to the toast element
+ * @param id the ID of the toast element
+ * @param prefix the prefix for the generated CSS class of the toast element resulting in the form ``$prefix-$hash``
+ * @param toastBuild a lambda expression for setting up the toast containing the alert
+ * @param build a lambda expression for setting up the alert component
+ */
+fun RenderContext.alertToast(
+    styling: BasicParams.() -> Unit = {},
+    baseClass: StyleClass? = null,
+    id: String? = null,
+    prefix: String = "alert",
+    toastBuild: ToastComponent.() -> Unit = { },
+    build: AlertComponent.() -> Unit,
+): SimpleHandler<Unit> {
+
+    val pendingToastStore = object : RootStore<AddToast>({ }) {
+        val show = handle {
+            showAlertToast(styling, baseClass, id, prefix, toastBuild, build)
+            it
+        }
+    }
+    return pendingToastStore.show
+}
+
+/**
+ * Creates and alert and displays it as a toast message.
+ * The toast's theme properties are automatically set but can manually be overridden by passing [toastBuild].
+ *
+ * @param styling a lambda expression for declaring the styling of the toast using fritz2's styling DSL
+ * @param baseClass optional CSS class that should be applied to the toast element
+ * @param id the ID of the toast element
+ * @param prefix the prefix for the generated CSS class of the toast element resulting in the form ``$prefix-$hash``
+ * @param toastBuild a lambda expression for setting up the toast containing the alert
+ * @param build a lambda expression for setting up the alert component
+ */
+fun RenderContext.showAlertToast(
+    styling: BasicParams.() -> Unit = {},
+    baseClass: StyleClass? = null,
+    id: String? = null,
+    prefix: String = "alert",
+    toastBuild: ToastComponent.() -> Unit = { },
+    build: AlertComponent.() -> Unit,
+) {
+    val alert = AlertComponent().apply(build)
+
+    showToast {
+        toastBuild()
+        content {
+            alert.show(
+                this,
+                styling = {
+                    styling()
+                    paddings { right { giant } }
+                },
+                baseClass,
+                id,
+                prefix
+            )
+        }
+        closeButtonStyle {
+            alert.variantStyles.text()
+        }
+    }
+}
+
+/**
+ * Convenience extension to display a [ComponentValidationMessage] as an alert.
+ * The alert's severity and content are determined from the validation message's properties.
+ *
+ * @param renderContext RenderContext to render the alert in.
+ */
+fun ComponentValidationMessage.asAlert(renderContext: RenderContext) {
+    renderContext.alert {
+        severity { when(severity) {
```

Comment:
Currently, Severity cannot be Success

Reply:
This is due to the fact, that ``Severity`` currently does not have a ``Success`` value. I have allready added this within my enhancements of formControl (PR#256), but that should be reflected here as well of course. So in order to be compliant here is the code from my branch that should be applied here as well:
```kotlin
enum class Severity {
    Info, Success, Warning, Error
}
```
(The order matters! So ``Success`` should be in second position!)

Reply:
Exactly my point, good to know it's already in progress.